### PR TITLE
feat: Add security checks page

### LINKED ISSUE #1665: Add Security Check page
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
The new [Security Check](https://github.com/parse-community/parse-server/issues/7246) feature of Parse Server currently writes weak security settings in the logs or returns them as JSON via REST request to `/security` endpoint.

- Human-reading the report in logs or as JSON is inconvenient
- Outputting the report in logs is already discouraged an regarded a weak security setting itself because it potentially exposes points of attack in the logs

### Feature / Enhancement Description
Add a Security Check page to Parse Dashboard that displays the report in a convenient UI.

The feature would send a request to the `/security` endpoint to receive the report in JSON format. The report schema is already described in https://github.com/parse-community/parse-server/pull/7247#issue-585913691. The page would display a table of that report, ideally with visual elements to highlight failed / successful security checks.

### Example Use Case
(none)

### Alternatives / Workarounds
Read reports in logs or manually via REST request.

### 3rd Party References
(none)

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description

There is currently no page to display security checks

Closes: #1665
### Approach

Screenshot 2023-06-29 at 6 11 39 pm

Screenshot 2023-06-29 at 6 11 51 pm

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by mtrezza** [comment]
Lovely! Do you think you could split this up into more columns?

| Check                 | Status | Issue                                    | Solution                                 |
|-----------------------|--------|------------------------------------------|------------------------------------------|
| <b>Overall Status</b> | ❌      |                                          |                                          |
| Secure master key     | ❌      | The Parse Server master key is insecure. | Choose a longer and/or more complex key. |
| Security log disabled          | ✅      |                                        |                                         |

File: src/dashboard/Settings/Security/Security.react.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    if (this.state.loading) {
+      return (
+        <LoaderContainer loading={this.state.loading}>
+          <div style={{ minHeight: '100vh' }} className={styles.content} />
+        </LoaderContainer>
+      );
+    }
+    return (
+      <div>
+        <Fieldset legend="Result">
```
Comment (mtrezza) [inline src/dashboard/Settings/Security/Security.react.js]:
```suggestion
        <Fieldset legend="Overall Security Status">
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
I'm not sure but for this feature to be implemented we may need to introduce a config option to hide the page per app or per user. It's exposing the security status of the server and its vulnerabilities. Any dashboard user can see this info easily, is that something of concern?

**Comment by dblythy** [comment]
`security.enableCheck` is disabled by default on the server, so this is what shows:

Screenshot 2023-06-30 at 1 13 45 pm

Perhaps we should pass this to `featuresRouter` and hide/show this page depending on the server settings?

**Comment by dblythy** [comment]
Related: https://github.com/parse-community/parse-server/issues/8678

**Comment by mtrezza** [comment]
That would mean that someone could not enable security checks if they didn't want to display possible attack vectors for all users in Parse Dashboard, right? If so, I'm not sure this would be an acceptable constraint. I think it should be possible to enable security checks on the server while hiding the results from the dashboard user.

If a dashboard user has access to the master key by inspecting the API requests anyway, it may matter less, but I still think this kind of information should not be displayed so prominently to everyone.

**Comment by dblythy** [comment]
Screenshot 2023-07-17 at 4 34 35 pm

**Comment by dblythy** [comment]
I'm not sure if it's currently possible to specify features per user (other than readonly). I think the feature has to be either on for all users, or off for all. We can add an option `enableSecurityChecks: true` that can only show security checks if enabled in the dashboard config.

And yes, any dashboard user can inspect any network request to read the masterKey.

**Comment by mtrezza** [comment]
Well, it's not that big of a deal then, but if you want to add an option to disable the security checks page in the dashboard config (for all apps or per app or per user, whatever is easiest), then I think it would be better. Otherwise I think we can merge it anyway.

Btw the new table layout looks nice! An additional luxury would be a refresh button, so developers can play around with the server settings and see the status change. Also just a suggestion, happy to merge without.

**Comment by dblythy** [comment]
Screenshot 2023-07-18 at 10 35 32 am

There is a reload button in the top right, I just missed it in the screenshot

**Comment by mtrezza** [comment]
Nice!

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-alpha.28](https://github.com/parse-community/parse-dashboard/releases/tag/5.2.0-alpha.28)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-beta.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0)

**Pull request merged**